### DISCUSSION TITLE
ESModules + ts-jest = Must use import to load ES Module: ****src/db/entities/****.entity.ts

### DISCUSSION THREAD

**Discussion by artsiommiksiuk** [mikro-orm/mikro-orm#5461] [General]
### Describe the bug

Using ESModules with ts-jest and trying to initialize mikro orm, causes autodiscovery to throw this with the error in the title.

Following the stack trace, I've stumbled to this:
https://github.com/mikro-orm/mikro-orm/blob/7fc4354589d98b632a274d32114c3607d22f8818/packages/core/src/utils/Utils.ts#L1072

I removed ```process.env.TS_JEST``` check and discovery passed and tests were executed.

I'm assuming there should be additional check, that it's not only TS_JEST env, but also that project is not in ESM mode.

### Reproduction

I have quite complex setup, not sure I can reproduce.

Here's stack trace:

```
Test suite failed to run

    Must use import to load ES Module: /Users/****/Projects/backend/src/db/entities/****.entity.ts

       5 |
       6 | export async function initOrm () {
    >  7 |     const orm = await MikroORM.init({
         |                 ^
       8 |         ...baseOptions,
       9 |
      10 |         metadataCache: { enabled: false },

      at Runtime.requireModule (../node_modules/jest-runtime/build/index.js:850:21)
      at Function.dynamicImport (../node_modules/@mikro-orm/core/utils/Utils.js:861:20)
      at MetadataDiscovery.getEntityClassOrSchema (../node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:1200:45)
      at MetadataDiscovery.discoverDirectories (../node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:197:40)
      at MetadataDiscovery.discover (../node_modules/@mikro-orm/core/metadata/MetadataDiscovery.js:43:9)
      at MikroORM.discoverEntities (../node_modules/@mikro-orm/core/MikroORM.js:172:25)
      at Function.init (../node_modules/@mikro-orm/core/MikroORM.js:44:9)
      at initOrm (db/test.config.ts:7:17)
      at domain/****.test.ts:13:13
```

### What driver are you using?

@mikro-orm/sqlite

### MikroORM version

6.2.0

### Node.js version

20.11.1

### Operating system

MacOS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Hmm, I guess I completely forgot people still use jest even with ESM projects - that's because it was nearly impossible to set up some time ago :D (also, vitest just works - use that if your project is ESM, and thank me later! although that would also mean you need ts-morph to get the decorator metadata, or specify types explicitly) 

Let's see if it won't backfire, removed the whole `require` block (7e8409bb86ca6a35c755700b77e17b425c5ab186) as it shouldn't be needed in the first place, not even for JSON files (and the ORM test suite is passing just fine without it).

**Reply by artsiommiksiuk** [reply]
Oh, I figured I forgot to answer.

Thanks for the quick fix and advice!

That's all I wanted to say 🙂

**Reply by artsiommiksiuk** [reply]
@B4nan Returning here to thank you - migrated to vitest 😃
